packages/connect-node/src/grpc-http2-transport.ts

```diff
@@ -163,33 +212,44 @@ export function createGrpcHttp2Transport(
 
             const headersPromise = readResponseHeader(stream);
             const trailerPromise = readResponseTrailer(stream);
-            const envelope = encodeEnvelope(
-              messageFlag,
-              serialize(req.message)
-            );
-            await write(stream, envelope);
+            const enveloped = encodeEnvelope(flag, requestBody);
+            await write(stream, enveloped);
             await end(stream);
 
             const [responseCode, responseHeader] = await headersPromise;
-            grpcValidateResponse(useBinaryFormat, responseCode, responseHeader);
-
+            const { compression } = grpcValidateResponseWithCompression(
+              useBinaryFormat,
+              acceptCompression,
+              responseCode,
+              responseHeader
+            );
             const messageResult = await readEnvelope(stream);
             const eofResult = await readEnvelope(stream);
+
             if (!eofResult.done) {
               throw "extraneous data";
             }
             const trailer = await trailerPromise;
             grpcValidateTrailer(trailer);
+
             if (messageResult.done) {
               throw "premature eof";
             }
 
+            let responseBody = messageResult.value.data;
+            if (compression) {
+              responseBody = await compression.decompress(
+                responseBody,
+                readMaxBytes
+              );
+            }
```

Comment:
See comments on the grpc http transport. I think all points apply here as well, and these lines should be identical between the two transports.